internal/reader/home_chain.go

```diff
@@ -111,19 +111,38 @@ func (r *homeChainPoller) poll() {
 }
 
 func (r *homeChainPoller) fetchAndSetConfigs(ctx context.Context) error {
-	var chainConfigInfos []ChainConfigInfo
-	err := r.homeChainReader.GetLatestValue(
-		ctx, "CCIPConfig", "getAllChainConfigs", nil, &chainConfigInfos,
-	)
-	if err != nil {
-		return err
+	var allChainConfigInfos []ChainConfigInfo
+	pageIndex := uint64(0)
+	pageSize := uint64(500)
+
+	for {
+		var chainConfigInfos []ChainConfigInfo
+		err := r.homeChainReader.GetLatestValue(
+			ctx,
+			"CCIPConfig",
+			"getAllChainConfigs",
+			map[string]interface{}{
+				"pageIndex": pageIndex,
+				"pageSize":  pageSize,
+			},
+			&chainConfigInfos,
+		)
+		if err != nil {
```

Comment:
q: when do we get an error? Is it always guaranteed to return an empty array if I am accessing out of bounds pages (e.g. 10 pages beyond what is available)?

Should we add a comment that we always expect a result to be returned?

Reply:
it returns anempty array in absence of any data, will never throw errors
there could be scenarios due to rpc issues